common/config/src/argv.rs

```diff
@@ -0,0 +1,719 @@
+        }
+
+        if self.lookback.is_some() {
+            raw.log.lookback = self.lookback.map(|v| v.to_string());
+        }
+
+        if self.use_k8s_enrichment.is_some() {
+            raw.log.use_k8s_enrichment = self.use_k8s_enrichment.map(|v| v.to_string());
+        }
+
+        if self.log_k8s_events.is_some() {
+            raw.log.log_k8s_events = self.log_k8s_events.map(|v| v.to_string());
+        }
+
+        if self.db_path.is_some() {
+            raw.log.db_path = self.db_path.map(PathBuf::from);
+        }
+
+        if !self.line_exclusion.is_empty() {
+            let regex = raw.log.line_exclusion_regex.get_or_insert(Vec::new());
+            with_csv(self.line_exclusion)
+                .iter()
+                .for_each(|v| regex.push(v.clone()));
+        }
+
+        if !self.line_inclusion.is_empty() {
+            let regex = raw.log.line_inclusion_regex.get_or_insert(Vec::new());
+            with_csv(self.line_inclusion)
+                .iter()
+                .for_each(|v| regex.push(v.clone()));
+        }
+
+        if !self.line_redact.is_empty() {
+            let regex = raw.log.line_redact_regex.get_or_insert(Vec::new());
+            with_csv(self.line_redact)
+                .iter()
+                .for_each(|v| regex.push(v.clone()));
+        }
+
+        raw
+    }
+
+    /// Parse command line options, default env vars and additional (deprecated) env vars.
+    pub fn from_args_with_all_env_vars() -> ArgumentOptions {
+        let options: ArgumentOptions = ArgumentOptions::from_args();
+        ArgumentOptions::parse_deprecated(options)
+    }
+
+    fn parse_deprecated(options: ArgumentOptions) -> ArgumentOptions {
+        let mut options = options;
+        if let Ok(v) = env_var(env::INGESTION_KEY_ALTERNATE) {
+            // Do not warn about alternate name for key
+            options.key = Some(v);
+        }
+
+        macro_rules! deprecated_env {
+            ($key: ident, $var_name: ident, Option<$ftype: ty>) => {
+                if let Ok(v) = env_var(env::$var_name) {
+                    if let Ok(parsed) = std::str::FromStr::from_str(&v) {
+                        options.$key = Some(parsed);
+                        info!("Using deprecated env var '$var_name'");
+                    } else {
+                        warn!("Deprecated env var '$var_name' could not be parsed to $ftype");
+                    }
+                }
+            };
+            ($key: ident, $var_name: ident, $ftype: ty) => {
+                if let Ok(v) = env_var(env::$var_name) {
+                    if let Ok(parsed) = std::str::FromStr::from_str(&v) {
+                        options.$key = parsed;
+                        info!("Using deprecated env var '$var_name'");
+                    } else {
+                        warn!("Deprecated env var '$var_name' could not be parsed to $ftype");
+                    }
+                }
+            };
```

Comment:
My macro fu is weak, there's probably a way to avoid repetition for these two patterns.

Reply:
There are other ways to write it, but I don't think there's a way to make it much shorter, you have to decide between Option and non-option no matter what